geps/gep-3779/index.md

```diff
@@ -42,20 +37,432 @@ Kubernetes core provides NetworkPolicies as one way to do it. Network Policies h
+* Implementations **should not** adopt `LabelSelector` targeting in their own custom policies attached to Gateway API resources until the pattern is sufficiently battle-tested and promoted to a standard feature. This staged approach mitigates risks of ecosystem fragmentation.
+
+Here is how it is going to look like:
+
+```go
+
+// PolicyTargetReferenceWithLabelSelectors specifies a reference to a set of Kubernetes
+// objects by Group and Kind, with an optional label selector to narrow down the matching
+// objects.
+//
+// Currently, we only support label selectors when targeting Pods.
+// This restriction is intentional to limit the complexity and potential
+// ambiguity of supporting label selectors for arbitrary Kubernetes kinds.
+// Unless there is a very strong justification in the future, we plan to keep this
+// functionality limited to selecting Pods only.
+//
+// This is currently experimental in the Gateway API and should only be used
+// for policies implemented within Gateway API. It is currently not intended for general-purpose
+// use outside of Gateway API resources.
+// +kubebuilder:validation:XValidation:rule="!(has(self.selector)) || (self.kind == 'Pod' && (self.group == '' || self.group == 'core'))",message="Selector may only be set when targeting Pods."
+type PolicyTargetReferenceWithLabelSelectors struct {
+  // Group is the group of the target object.
+  Group Group `json:"group"`
+
+  // Kind is the kind of the target object.
+  Kind Kind `json:"kind"`
+
+  // Selector is the label selector of target objects of the specified kind.
+  Selector *metav1.LabelSelector `json:"selector"`
+}
+
+```
+
+##### Enhanced Discoverability with `gwctl`
+
+A key challenge with `LabelSelector` is the loss of discoverability. It’s easier to see which policies target a `Service` but difficult to determine which policies might affect a specific pod.
+
+To address this, **investment in tooling is required.** Specifically, the `gwctl` CLI tool should be enhanced to provide insights such as:
+
+```sh
+gwctl describe pods pod1 
+
+...
+InheritedPolicies:
+  Type                   Name                                 Target Kind   Target Name/Expression
+  ----                   ----                                 -----------   -----------
+  AuthorizationPolicy  demo-authz-policy-on-payment-pods      Pod           Selector={foo: a, bar: 2}
+...
+
+# List all AuthorizationPolicy resources in json format in the active namespace
+gwctl get authorizationPolicy -o json
+
+```
+
+Without dedicated tooling, the `LabelSelector` approach could significantly degrade the user experience and observability.
+
+##### Namespace-Wide Policies
+
+A common case for authorization is to target a whole namespace. We can achieve the above with a few patterns:
+
+###### Option 1
+
+Target a namespace, name MUST be empty (as this is already namespaced policy)
```

Comment:
As described, `targetRef` doesn't include `name` -- did you intend to include it? 🙂 

Part of the reason I ask is experience with trust-manager, which _only_ allows selecting namespaces using label selectors, which is incredibly annoying if you just want to say "put the trust bundle in the `linkerd` namespace". Targeting a namespace by name makes sense; only allowing a label selector doesn't really help much when your target is a single namespace.

Reply:
Good catch about name.

>only allowing a label selector doesn't really help much when your target is a single namespace.

Not sure I understand, can you elaborate here? Targeting a namespace by name is rather confusing, as the policy is namespace scoped, it can not possibly target a namespace thats not the namespace it is deployed in.

empty label selector, while not perfect, signals "everything in this namespace"

Reply:
added clarifications, PTAL